File: src/main/java/no/nav/fo/forenkletdeploy/rest/JiraIssueResource.java
package no.nav.fo.forenkletdeploy.rest;

import no.nav.fo.forenkletdeploy.domain.JiraIssue;
import no.nav.fo.forenkletdeploy.domain.JiraIssues;
import no.nav.fo.forenkletdeploy.service.JiraIssueService;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.inject.Inject;
import java.util.List;

@RestController
@RequestMapping("/api/jira")
public class JiraIssueResource {
    private final JiraIssueService jiraIssueService;

    @Inject
    JiraIssueResource(JiraIssueService jiraIssueService) {
        this.jiraIssueService = jiraIssueService;
    }

    @GetMapping("/")
    public List<JiraIssue> getAllFoIssues() {
        return jiraIssueService.getUserStories();
    }

    @GetMapping("/{issueid}")
    public JiraIssue getJiraIssue(@PathVariable("issueid") String issueid) {
        return jiraIssueService.getUserStory(issueid)
                .withKey(issueid);
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/rest/JiraIssueResource.kt
package no.nav.fo.forenkletdeploy.rest

import no.nav.fo.forenkletdeploy.service.JiraService
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

import javax.inject.Inject

@RestController
@RequestMapping("/api/jira")
class JiraIssueResource @Inject
constructor(val jiraService: JiraService) {
    val LOG = LoggerFactory.getLogger(JiraIssueResource::class.java)

    @GetMapping
    fun getJiraIssues(@RequestParam(name = "issue") issues: List<String>) =
            issues
                    .filter { !it.startsWith("kaizen", ignoreCase = true) }
                    .mapNotNull {
                        try {
                            jiraService.getUserStory(issueId = it)
                        } catch (ex: Exception) {
                            LOG.error("Kunne ikke finne jira-issue $it", ex)
                            null
                        }
                    }

    @GetMapping("/{issueid}")
    fun getJiraIssue(@PathVariable("issueid") issueid: String) =
            jiraService.getUserStory(issueid).copy(key = issueid)
}
